python/ray/air/tests/test_tensor_extension.py

```diff
@@ -178,7 +178,14 @@ def test_arrow_variable_shaped_tensor_array_slice():
         slice(0, 3),
     ]
     for slice_ in slices:
-        for o, e in zip(ata[slice_], arr[slice_]):
+        ata_slice = ata[slice_]
+        ata_slice_np = ata_slice.to_numpy()
+        arr_slice = arr[slice_]
+        # Check for equivalent dtypes and shapes.
+        assert ata_slice_np.dtype == arr_slice.dtype
```

Comment:
Why doesn't `ArrowTensorArrow` expose `dtype` and `shape` methods like `TensorArray`?

Reply:
Both the data type and element shape are exposed on the underlying tensor array type, i.e. `ata.type.storage.value_type` for the Arrow type, `ata.type.to_pandas_dtype().numpy_dtype` for the NumPy dtype, and `ata.type.shape` for the element shape.

We haven't exposed them directly on `ArrowTensorArray` under the assumption that the user will typically be working with the Pandas-side `TensorArray` extension type (if they're interacting with tensor extension types at all; they're hidden by default), so the Arrow side isn't optimized for end-user interaction.

Agreed that making the extension type more user-friendly is worth doing though.